### PR TITLE
Logical-spectralog

### PR DESCRIPTION
**Pull request by AlbertLee125**
Added the documentation of the Logical Spectralog. Added the original sources of the references.
And explain where the String IO refer to.

### CODE REVIEW

**Review by ZedongPeng — APPROVED** [review]

File: gdplib/logical/spectralog.py [hunk 1]
```diff
@@ -15,13 +15,35 @@
 from pyomo.environ import *
```
Comment (bernalde) [inline gdplib/logical/spectralog.py:13]: Are any of the commented lines above still relevant? Please modify accordingly

File: gdplib/logical/spectralog.py [hunk 2]
```diff
@@ -33,68 +55,152 @@ def build_model():
...
+            )
             * (m.C[k, j] / 100 - sum(m.P[k, i] * m.A[i, j] for i in m.wave_number))
             for k in m.compounds
         )
 
     m.profit = Objective(
-        expr=sum(m.val[j] for j in m.spectra_data) + 2 * sum(m.ent[k, i] for k in m.compounds for i in m.wave_number))
+        expr=sum(m.val[j] for j in m.spectra_data)
+        + 2 * sum(m.ent[k, i] for k in m.compounds for i in m.wave_number),
+        doc='Objective to maximize spectroscopic agreement and encourage compound presence.',
```
Comment (bernalde) [inline gdplib/logical/spectralog.py]: This description is off. Compare it to the comment below

**Review by bernalde — CHANGES_REQUESTED** [review]

File: gdplib/logical/spectralog.py (hunk 2 above)
Comment (AlbertLee125) [reply gdplib/logical/spectralog.py]: I changed the doc to make better alignment with the comment below.

File: gdplib/logical/spectralog.py (hunk 1 above)
Comment (AlbertLee125) [reply gdplib/logical/spectralog.py:13]: Modified the commented line as the header.

**Review by bernalde — APPROVED** [review]

**Pull request merged**